### PR TITLE
More tests for Vault using mocks.

### PR DESCRIPTION
**Pull request by lloydjm77**

### CODE REVIEW

File: pkg/vault/vault_test.go [hunk 1]
```diff
@@ -55,3 +59,110 @@ func TestLoadTokenExpectRetry(t *testing.T) {
 		}
 	}
 }
+
+// TestCreateCert tests that given a Vault client and test server, we can successfully
+// obtain a certificate with the requested options (CN, TTL, etc).
+func TestCreateCert(t *testing.T) {
```
Comment (alexdicianu) [inline pkg/vault/vault_test.go:65]: I don't think we use CreateCert for anything in this repo. JR does that.

**Review by alexdicianu — APPROVED** [review]
I kinda liked the in memory test server approach but since it seems like it's more of a hack then let's mock.

File: pkg/vault/vault_test.go (hunk 1 above)
Comment (lloydjm77) [reply pkg/vault/vault_test.go:65]: We don't.  I just left it in there anyway.

**Comment by lloydjm77** [comment]
> I kinda liked the in memory test server approach but since it seems like it's more of a hack then let's mock.

I didn't have any problem with it other than every time we tried to upgrade the Vault APIs, it would break because of using that package.  Plus, the Vault guys said it was only intended to be used by their team for internal testing.

https://github.com/hashicorp/vault/issues/13386

File: pkg/vault/vault_test.go (hunk 1 above)
Comment (lloydjm77) [reply pkg/vault/vault_test.go:65]: I'll take it out.

**Pull request merged**